src/Space.php

```diff
@@ -171,20 +179,31 @@ public static function newFromKey( string $nsKey ) {
 	 */
 	private static function newFromRow( array $condition ) {
 		$loadBalancer = MediaWikiServices::getInstance()->getDBLoadBalancer();
+		$fields = [
+			'namespace_id',
+			'namespace_key',
+			'namespace_name',
+			'description',
+			'creator_id',
+			'archived'
+		];
 
 		if ( method_exists( $loadBalancer, 'getConnection' ) ) {
-			$dbr = MediaWikiServices::getInstance()->getDBLoadBalancer()->getConnection( DB_REPLICA );
+			$dbr = $loadBalancer->getConnection( DB_REPLICA );
 		} else {
-			$dbr = MediaWikiServices::getInstance()->getDBLoadBalancer()->getConnectionRef( DB_REPLICA );
+			$dbr = $loadBalancer->getConnectionRef( DB_REPLICA );
 		}
 
 		// It might happen that this function is called during run of update.php,
 		// while database is not property set up. In that case, give a sensible return value.
 		if ( !$dbr->tableExists( 'wss_namespaces', __METHOD__ ) ) {
 			return false;
 		}
+		if ( $dbr->fieldExists( 'wss_namespaces', 'protected', __METHOD__ ) ) {
+			$fields []= 'protected';
```

Comment:
$fields[] = "protected"
Do not include whitespace between variable expression and array offset